[3006.x] file.replace and file.search work properly with /proc files

Akm0d opened the pull request:
### What does this PR do?
This PR fixes the issue where file.replace and file.search do not work with /proc files in Python 3. The issue was caused by attempting to join bytes objects, which is not allowed in Python 3. The fix involves correctly handling bytes when reading from files in the /proc directory.

### What issues does this PR fix or reference?
Fixes #63102

### Previous Behavior
Attempting to use file.replace or file.search on files in the /proc directory resulted in a TypeError because the code tried to join bytes objects, which is not allowed in Python 3.

### New Behavior
The code now correctly handles bytes when reading from files in the /proc directory, preventing the TypeError and ensuring that file.replace and file.search work as expected.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Akm0d committed:
file.replace and file.search work properly with /proc files

dmurphy18 commented on salt/modules/file.py:
```diff
@@ -2539,7 +2539,7 @@ def replace(
                 r_data = mmap.mmap(r_file.fileno(), 0, access=mmap.ACCESS_READ)
             except (ValueError, OSError):
                 # size of file in /proc is 0, but contains data
-                r_data = salt.utils.stringutils.to_bytes("".join(r_file))
```
I thought the standard was to make use of salt.utils etc rather than doing 'b'.

dmurphy18 reviewed (commented):
Wondering about by-passing salt.utils, remember Erik taking me to task for by-passing them, has best practice changed for Salt code ????

Akm0d commented:
Salt utils is no longer necessary here.  This part of the code hasn't been working since we switched to Python3.  We already have the encoded byte data from opening the file with "rb".   

I've changed the code so that it joins on an empty bytestring rather than joining bytes to an empty unicode string then using salt utils to re-convert it back to bytes.

Salt utils is not necessary here unless there are multiple ways to encode an empty byestring.
